server/websocket_test.go

```diff
@@ -960,4 +961,121 @@ func TestHandleJoin(t *testing.T) {
 		require.Empty(t, p.addSessionsBatchers)
 		p.mut.RUnlock()
 	})
+
+	t.Run("admin warning", func(t *testing.T) {
+		channelID := model.NewId()
+		userID := model.NewId()
+		connID := model.NewId()
+		postID := model.NewId()
+		authSessionID := ""
+
+		os.Setenv("MM_CALLS_CONCURRENT_SESSIONS_THRESHOLD", "1")
+		defer os.Unsetenv("MM_CALLS_CONCURRENT_SESSIONS_THRESHOLD")
+
+		mockAPI.On("HasPermissionToChannel", userID, channelID, model.PermissionCreatePost).Return(true).Once()
+		mockAPI.On("GetChannel", channelID).Return(&model.Channel{
+			Id: channelID,
+		}, nil).Twice()
+
+		mockAPI.On("GetChannelStats", channelID).Return(&model.ChannelStats{
+			MemberCount: 1,
+		}, nil).Once()
+
+		mockAPI.On("HasPermissionTo", userID, model.PermissionManageSystem).Return(false).Once()
+
+		// Call lock
+		mockAPI.On("KVSetWithOptions", "mutex_call_"+channelID, []byte{0x1}, mock.Anything).Return(true, nil)
+
+		// We'd be starting a new call
+		mockMetrics.On("IncWebSocketEvent", "out", wsEventCallHostChanged).Once()
+		mockAPI.On("PublishWebSocketEvent", wsEventCallHostChanged, mock.Anything,
+			&model.WebsocketBroadcast{UserId: userID, ChannelId: channelID, ReliableClusterSend: true}).Once()
+
+		// Call started post creation
+		mockAPI.On("GetUser", userID).Return(&model.User{Id: userID}, nil).Once()
+		mockAPI.On("GetConfig").Return(&model.Config{}, nil).Times(3)
+		mockAPI.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(&model.Post{Id: postID}, nil).Once()
```

Comment:
Whichever one is the admin warning, that's what you'd want to run an assert on.

Reply:
@cpoile Is the assertion just a way to avoid the panic, or is there more to it?

Reply:
It's what I clarified above -- asserting that we're actually doing the CreatePost.